interfaces/builtin/basedeclaration.go

```diff
@@ -267,6 +267,11 @@ slots:
       slot-snap-type:
         - core
     deny-auto-connection: true
+  lxd-client:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: false
```

Comment:
Not sure what that flag means, but if it means that other snaps will be allowed to auto-connect without special casing in the store, then this is a terrible idea.

Accessing the LXD socket is equivalent to having root access on the system. It lets you access everything on the filesystem and effectively bypass all seccomp and apparmor policies.

It's certainly fine for us to hardcode a few exceptions for well behaved, trusted snaps, but we should never allow a random snap to the store to connect to LXD without an explicit "snap connect" from the user.